### PR TITLE
Cosmos docstring review

### LINKED ISSUE #7788: [Cosmos] Review all docstrings

### LINKED ISSUE #8624: Flaky Emulator Tests
There are a handful of flaky Cosmos Emulator tests that are currently failing unrelated PRs (even after a recent fix for emulator/live discrepancies was recently merged, AFAIK). 

Specifically: https://github.com/Azure/azure-sdk-for-python/pull/8607 and https://github.com/Azure/azure-sdk-for-python/pull/8548 at least are not passing due to Cosmos emulator tests. 

The tests often seem to be these query and retry tests:

```

self = <query_tests.QueryTest testMethod=test_max_item_count_honored_in_order_by_query>

    def test_max_item_count_honored_in_order_by_query(self):
        created_collection = self.config.create_multi_partition_collection_with_custom_pk_if_not_exist(self.client)
        docs = []
        for i in range(10):
            document_definition = {'pk': 'pk', 'id': 'myId' + str(uuid.uuid4())}
            docs.append(created_collection.create_item(body=document_definition))
    
        query = 'SELECT * from c ORDER BY c._ts'
        query_iterable = created_collection.query_items(
            query=query,
            max_item_count=1,
            enable_cross_partition_query=True
        )
>       self.validate_query_requests_count(query_iterable, 15 * 2 + 1)

test\query_tests.py:193: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test\query_tests.py:210: in validate_query_requests_count
    self.assertEqual(self.count, expected_count)
E   AssertionError: 33 != 31
_________ Test_retry_policy_tests.test_default_retry_policy_for_query _________

self = <retry_policy_tests.Test_retry_policy_tests testMethod=test_default_retry_policy_for_query>

    def test_default_retry_policy_for_query(self):
        document_definition_1 = { 'id': 'doc1',
                                  'name': 'sample document',
                                  'key': 'value'}
        document_definition_2 = { 'id': 'doc2',
                                  'name': 'sample document',
                                  'key': 'value'}
    
        self.created_collection.create_item(body=document_definition_1)
        self.created_collection.create_item(body=document_definition_2)
    
        try:
            original_execute_function = _retry_utility.ExecuteFunction
            mf = self.MockExecuteFunctionConnectionReset(original_execute_function)
            _retry_utility.ExecuteFunction = mf
    
            docs = self.created_collection.query_items(query="Select * from c", max_item_count=1, enable_cross_partition_query=True)
    
            result_docs = list(docs)
            self.assertEqual(result_docs[0]['id'], 'doc1')
            self.assertEqual(result_docs[1]['id'], 'doc2')
    
            # TODO: Differing result between live and emulator
            if 'localhost' in self.host or '127.0.0.1' in self.host:
>               self.assertEqual(mf.counter, 12)
E               AssertionError: 18 != 12

test\retry_policy_tests.py:198: AssertionError
```

@annatisch suggested

> I wonder if we can mark these as xfailures so as not to break CI for everyone

Which I am inclined to agree is worth considering. But this is also probably worth a wider discussion. 

cc @mayurid @srinathnarayanan thoughts? Can we just mark these `xfail` for now?

### PR DESCRIPTION
**Pull request by bryevdv**
issues: closes #7788 closes #8624 

cc @annatisch 

Proposed to move documentation of "common kwargs" to a separate issue/PR.

A few comments:

* I'm old fashioned and still read often docstrings in a terminal. I definitely appreciate when text/paras in docstrings are < ~80 cols, so I have tried to hew to that. 

* My perception is that the preference in the repo is for args lists that are vertically compact, so I have made some edits around that as well.

* Besides typo, grammar fixes, I tried to implement parallel phrasing in places where it was simple and made sense.

* Added notes about some exceptions up front (e.g. API for deleting a non-existent item could go either way, as a user I'd like to know immediately I need to basically always wrap with a `try/except`)

It would be an interesting exercise to review use of tense and number in docstrings across all the projects. I am personally in the *(Calling this function will) **"Update a specified document widget"*** camp, and not the *(A call to this function) **"Updates the specified document widget"*** camp.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/cosmos/azure-cosmos/azure/cosmos/_routing/collection_routing_map.py [hunk 1]
```diff
@@ -120,10 +116,8 @@ def get_range_by_partition_key_range_id(self, partition_key_range_id):
     def get_overlapping_ranges(self, provided_partition_key_ranges):
         """Gets the partition key ranges overlapping the provided ranges
 
-        :param list provided_partition_key_ranges:
-            List of partition key ranges.
-        :return:
-            List of partition key ranges, where each is a dict.
+        :param list provided_partition_key_ranges:cList of partition key ranges.
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_routing/collection_routing_map.py]: I mischievous "c"

File: sdk/cosmos/azure-cosmos/azure/cosmos/_session.py [hunk 2]
```diff
@@ -205,9 +195,11 @@ def parse_session_token(response_headers):
 
 
 class Session(object):
-    """
-    State of a Azure Cosmos session. This session object
-    can be shared across clients within the same process
+    """State of a Azure Cosmos session.
+
+    This session object can be shared across clients within the same process
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_session.py]: Final period.

Comment (annatisch) [reply sdk/cosmos/azure-cosmos/azure/cosmos/_session.py]: Actually periods missing throughout this module.

File: sdk/cosmos/azure-cosmos/azure/cosmos/container.py [hunk 3]
```diff
@@ -42,20 +42,18 @@
...
+
+    This class should not be instantiated directly. Instead, use the
+    :func:`DatabaseProxy.get_container_client` method to get an existing
+    container, or the :func:`Database.create_container` method to create a
+    new container.
 
-    A container in an Azure Cosmos DB SQL API database is a collection of documents,
-    each of which represented as an Item.
+    A container in an Azure Cosmos DB SQL API database is a collection of
+    documents, each of which represented as an Item.
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/container.py]: each of which _is_ represented as an Item.

File: sdk/cosmos/azure-cosmos/azure/cosmos/database.py [hunk 4]
```diff
@@ -42,20 +42,22 @@
...
-    This class should not be instantiated directly, use :func:`CosmosClient.get_database_client` method.
+    """An interface to interact with a specific database.
+
+    This class should not be instantiated directly. Instead use the
+    :func:`CosmosClient.get_database_client` method.
 
     A database contains one or more containers, each of which can contain items,
     stored procedures, triggers, and user-defined functions.
 
     A database can also have associated users, each of which configured with
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/database.py]: each of which [is, can be] configured....

File: sdk/cosmos/azure-cosmos/azure/cosmos/documents.py [hunk 5]
```diff
@@ -298,7 +299,7 @@ def __init__(self):
 
 
 class ProxyConfiguration(object):
-    """Configurations for proxy.
+    """Configurations for a proxy.
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/documents.py]: Configuration singular

File: sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py [hunk 6]
```diff
@@ -19,9 +19,11 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+"""Create partition keys in the Azure Cosmos DB SQL API service.
+"""
 
 class NonePartitionKeyValue(object):
-    """Represents none value for partitionKey when it's missing in a containers.
+    """Represents None value for partitionKey when it's missing in a containers.
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py]: Either "container" singular, or "in containers"

File: sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py [hunk 7]
```diff
@@ -38,14 +40,14 @@ class _Undefined(object):
...
 
     See https://docs.microsoft.com/azure/cosmos-db/partitioning-overview#choose-partitionkey
-    for more information on how to choose partition keys.
+    for information on how to choose partition keys.
 
     :ivar path: The path of the partition key
-    :ivar kind: What kind of partition key is being defined
-    :ivar version: The version of the partition key
+    :ivar kind: What kind of partition key is being defined (default: "Hash")
+    :ivar version: The version of the partition key (defatul: 2)
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py]: Typo: "default"

Comment (bryevdv) [reply sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py]: fixed in https://github.com/Azure/azure-sdk-for-python/pull/8607/commits/c14bb0d8adcb5e63ac1d6a2fab3819ccca50707e

File: sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py (hunk 6 above)
Comment (bryevdv) [reply sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py]: fixed in https://github.com/Azure/azure-sdk-for-python/pull/8607/commits/c14bb0d8adcb5e63ac1d6a2fab3819ccca50707e

File: sdk/cosmos/azure-cosmos/azure/cosmos/documents.py (hunk 5 above)
Comment (bryevdv) [reply sdk/cosmos/azure-cosmos/azure/cosmos/documents.py]: fixed in https://github.com/Azure/azure-sdk-for-python/pull/8607/commits/c14bb0d8adcb5e63ac1d6a2fab3819ccca50707e

File: sdk/cosmos/azure-cosmos/azure/cosmos/database.py (hunk 4 above)
Comment (bryevdv) [reply sdk/cosmos/azure-cosmos/azure/cosmos/database.py]: fixed in https://github.com/Azure/azure-sdk-for-python/pull/8607/commits/c14bb0d8adcb5e63ac1d6a2fab3819ccca50707e

File: sdk/cosmos/azure-cosmos/azure/cosmos/container.py (hunk 3 above)
Comment (bryevdv) [reply sdk/cosmos/azure-cosmos/azure/cosmos/container.py]: fixed in https://github.com/Azure/azure-sdk-for-python/pull/8607/commits/c14bb0d8adcb5e63ac1d6a2fab3819ccca50707e

File: sdk/cosmos/azure-cosmos/azure/cosmos/_session.py (hunk 2 above)
Comment (bryevdv) [reply sdk/cosmos/azure-cosmos/azure/cosmos/_session.py]: fixed in https://github.com/Azure/azure-sdk-for-python/pull/8607/commits/c14bb0d8adcb5e63ac1d6a2fab3819ccca50707e

File: sdk/cosmos/azure-cosmos/azure/cosmos/_routing/collection_routing_map.py (hunk 1 above)
Comment (bryevdv) [reply sdk/cosmos/azure-cosmos/azure/cosmos/_routing/collection_routing_map.py]: fixed in https://github.com/Azure/azure-sdk-for-python/pull/8607/commits/c14bb0d8adcb5e63ac1d6a2fab3819ccca50707e

**Pull request merged**